packages/datadog-plugin-mocha/src/index.js

```diff
@@ -1,25 +1,42 @@
-const { promisify } = require('util')
+'use strict'
+
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
 
-const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
-const { SAMPLING_PRIORITY, SPAN_TYPE, RESOURCE_NAME } = require('../../../ext/tags')
-const { AUTO_KEEP } = require('../../../ext/priority')
 const {
+  CI_APP_ORIGIN,
   TEST_TYPE,
   TEST_NAME,
   TEST_SUITE,
+  TEST_FRAMEWORK_VERSION,
   TEST_STATUS,
   TEST_PARAMETERS,
-  TEST_FRAMEWORK_VERSION,
-  CI_APP_ORIGIN,
-  getTestEnvironmentMetadata,
-  getTestParametersString,
   finishAllTraceSpans,
+  getTestEnvironmentMetadata,
+  getTestSuitePath,
   getTestParentSpan,
-  getTestSuitePath
+  getTestParametersString
 } = require('../../dd-trace/src/plugins/util/test')
+const { SPAN_TYPE, RESOURCE_NAME, SAMPLING_PRIORITY } = require('../../../ext/tags')
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+const { AUTO_KEEP } = require('../../../ext/priority')
 
 const skippedTests = new WeakSet()
 
+function getAllTestsInSuite (root) {
```

Comment:
Can this be done on the publisher side so that the subscriber doesn't need to know about the internals? Or at least normalized.

Reply:
For example, if the implementation of Mocha changes, the subscriber shouldn't need to change.